lib/utils/device/deploy.ts

```diff
@@ -554,6 +555,12 @@ function generateImageName(serviceName: string): string {
 	return `local_image_${serviceName}:latest`;
 }
 
+function getImageNameFromTask(task: BuildTask): string {
+	return !task.external && task.tag
+		? task.tag
+		: generateImageName(task.serviceName);
```

Comment:
Nice. 👍   This PR reminded me of the `makeImageName` function defined at https://github.com/balena-io/balena-cli/blob/v12.53.2/lib/utils/compose_ts.ts#L420-L430
How about centralising the image name generation logic (cloud and device) by **deleting** the `generateImageName` function from this file and importing `makeImageName` instead? It looks like this file already imports from `compose_ts`, so it might be just a matter of updating the import statement at the top of the file, and then making the call:
```ts
    makeImageName('local_image', task.serviceName, 'latest')
```

Reply:
Oh, nice idea, I'll check it out.

Reply:
Done! 👍